cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -1007,6 +1010,62 @@ func deleteAllPodsBoundToNode(ctx *controllerContext, nodeName string) error {
 	return utilerrors.NewAggregate(errs)
 }
 
+func markFailedAndDeletePodWithCondition(ctx *controllerContext, pod *v1.Pod) error {
+	const fieldManager = "GCPControllerManager"
+	condition := corev1apply.PodCondition().
+		WithType(v1.DisruptionTarget).
+		WithStatus(v1.ConditionTrue).
+		WithReason("DeletionByGCPControllerManager").
+		WithMessage(fmt.Sprintf("%s: node no longer exists", fieldManager)).
+		WithLastTransitionTime(metav1.Now())
+
+	// Based on pod gc controller (https://github.com/kubernetes/kubernetes/blob/17bf864c1fc20e4badc9a6e0659c5190310462e1/pkg/controller/podgc/gc_controller.go)
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+
+		// Extract the pod status as GCPControllerManager may or may not own the pod phase, if
+		// it owns the phase then we need to send the field back if the condition
+		// is added.
+		podApply, err := corev1apply.ExtractPodStatus(pod, fieldManager)
```

Comment:
I think now extract is no longer needed as the controller never owns fields other than the phase and the condition, and it always updates both of them.

Reply:
Please follow the same approach as https://github.com/kubernetes/kubernetes/pull/115056

Reply:
Done.